packages/bootstrap/extra/bin/bootstrap

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+import argparse
+import logging
+import os
+import sys
+
+
+from dcos_internal_utils import bootstrap
+from dcos_internal_utils import exhibitor
+
+
+log = logging.getLogger(__name__)
+
+
+def dcos_adminrouter(b, opts):
+    b.cluster_id('/var/lib/dcos/cluster-id')
+
+
+def dcos_signal(b, opts):
+    b.cluster_id('/var/lib/dcos/cluster-id')
+
+
+def dcos_oauth(b, opts):
+    b.generate_oauth_secret('/var/lib/dcos/dcos-oauth/auth-token-secret')
+
+
+def noop(b, opts):
+    return
+
+
+bootstrappers = {
+    'dcos-adminrouter': dcos_adminrouter,
+    'dcos-signal': dcos_signal,
+    'dcos-oauth': dcos_oauth,
+    'dcos-3dt': noop,
+    'dcos-marathon': noop,
+    'dcos-mesos-master': noop,
+    'dcos-cosmos': noop,
+}
+
+
+def main():
+    if os.getuid() != 0:
+        log.error('bootstrap must be run as root')
+        sys.exit(1)
+
+    opts = parse_args()
+
+    logging.basicConfig(format='[%(levelname)s] %(message)s', level='INFO')
+    log.setLevel(logging.DEBUG)
+
+    log.info('Clearing proxy environment variables')
+    os.environ.pop('HTTP_PROXY', None)
+    os.environ.pop('http_proxy', None)
+    os.environ.pop('HTTPS_PROXY', None)
+    os.environ.pop('https_proxy', None)
+    os.environ.pop('NO_PROXY', None)
+    os.environ.pop('no_proxy', None)
+
+    dirs = [
+        '/var/lib/dcos',
+        '/run/dcos',
+        '/var/lib/dcos/dcos-oauth',
+    ]
+    for d in dirs:
+        log.info('Preparing directory {}'.format(d))
+        os.makedirs(d, exist_ok=True)
+
+    exhibitor.wait(opts.master_count)
+
+    b = bootstrap.Bootstrapper(opts.zk)
+
+    for service in opts.services:
+        if service not in bootstrappers:
+            log.error('Unknown service: {}'.format(service))
+            sys.exit(1)
+        log.debug('bootstrapping {}'.format(service))
+        bootstrappers[service](b, opts)
+
+
+def parse_args():
+    if os.path.exists('/etc/mesosphere/roles/master'):
```

Comment:
nit: We should probably make pkgpanda contain a utility function to get the roles which were present when the last `pkgpanda activate` raun (in theory the list of roles could change)